src/storage/v2/durability/snapshot.cpp

```diff
@@ -127,6 +139,230 @@ struct BatchInfo {
+      int vertex_fd = open(res.snapshot_path.string().c_str(), O_RDONLY);
+      if (vertex_fd == -1) {
+        throw RecoveryFailure("Couldn't open vertex snapshot part!");
+      }
+      utils::OnScopeExit close_vertex_fd([&] {
+        close(vertex_fd);
+        std::error_code ec;
+        std::filesystem::remove(res.snapshot_path, ec);
+        if (ec) spdlog::warn("Couldn't remove temporary vertex snapshot part {}: {}", res.snapshot_path, ec.message());
+      });
+      // Use sendfile for efficient copying (zero-copy)
+      off_t offset = 0;
+      const auto size = res.snapshot_size - offset;
+      ssize_t bytes_sent = sendfile(snapshot.native_handle(), vertex_fd, &offset, size);
+      if (bytes_sent == -1 || bytes_sent != size) {
+        throw RecoveryFailure("Couldn't append vertex part to snapshot!");
+      }
+      snapshot.SetPosition(current_offset + size);
+    }
+  }
+}
+
+// Return at least one batch (at least 2 elements: start and end gid)
+auto Batch(auto &&acc, const uint64_t step) {
+  // Skiplist sizes can change, last thread will pick up any new elements. In the end, these will be skiped (MVCC)
+  const auto n_batches = (step < 1) ? 1 : (acc.size() + step - 1) / step;
+
+  if (n_batches < 2) {
+    // Single batch, nothing to do
+    return std::vector<int64_t>{0, kEnd};
+  }
+
+  std::vector<int64_t> batches(n_batches + 1, kEnd);  // start and end gids
+  batches[0] = 0;                                     // Always start from the lowest possible
+  int i = 0;
+  int batch_id = 1;
+  for (const auto &elem : acc) {
+    if (step == i++) {
+      batches[batch_id++] = elem.gid.AsInt();  // This batch's start ID and previous batch's end ID
+      i = 1;                                   // 1 on purpose, as the first element is already in the batch
+      // Check if we have enough batches
+      if (acc.size() < batch_id * n_batches) break;
+      if (batch_id == n_batches) break;
+    }
+  }
+  return batches;
+}
+
+void MultiThreadedWorkflow(utils::SkipList<Edge> *edges, utils::SkipList<Vertex> *vertices, auto &&store_edges,
+                           auto &&store_vertices, const uint64_t items_per_batch, uint64_t &offset_edges,
+                           uint64_t &offset_vertices, Encoder &snapshot, uint64_t &edges_count,
+                           uint64_t &vertices_count, std::vector<BatchInfo> &edge_batch_infos,
+                           std::vector<BatchInfo> &vertex_batch_infos, std::unordered_set<uint64_t> &used_ids,
+                           uint64_t thread_count) {
+  SafeTaskQueue tasks;
+  using task_results_t = std::vector<std::pair<SnapshotBatchRes, std::promise<bool>>>;
+
+  // Generate edge tasks
+  std::vector<int64_t> start_egid{};
+  task_results_t edge_res{};
+  if (edges != nullptr) {  // No edges skiplist <=> no properties on edges
+    start_egid = Batch(edges->access(), items_per_batch);
+    edge_res = task_results_t{start_egid.size() - 1};  // last element is an end marker
```

Comment:
use () brackets for this constructor instead of initializer list

Reply:
@as51340 I can give you good reasons to prefer {} over ()
- avoid accidental implicit conversions